server/block/smoker.go

```diff
@@ -0,0 +1,123 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// Smoker is a type of furnace that cooks food items, similar to a furnace, but twice as fast. It also serves as a
+// butcher's job site block.
+type Smoker struct {
```

Comment:
Same comments as those on `BlastFurnace` apply here, also wrt `NewSmoker`. Just fill out the fields with sensible defaults.

Reply:
Alright, changed.